src/dbnode/persist/fs/files.go

```diff
@@ -1012,13 +1012,34 @@ func IndexSnapshotFiles(filePathPrefix string, namespace ident.ID) (FileSetFiles
 
 // FileSetAt returns a FileSetFile for the given namespace/shard/blockStart/volume combination if it exists.
 func FileSetAt(filePathPrefix string, namespace ident.ID, shard uint32, blockStart time.Time, volume int) (FileSetFile, bool, error) {
+	var pattern string
+	// If this is the initial volume, then we need to check if files were written with the legacy file naming (i.e.
+	// without the volume index) so that we can properly locate the fileset.
+	if volume == 0 {
+		dir := ShardDataDirPath(filePathPrefix, namespace, shard)
+		isLegacy, err := isFirstVolumeLegacy(dir, blockStart, checkpointFileSuffix)
+		// NB(nate): don't propagate ErrCheckpointFileNotFound here as expectation is to simply return an
+		// empty FileSetFile if files do not exist.
+		if err == ErrCheckpointFileNotFound {
+			return FileSetFile{}, false, nil
+		} else if err != nil && err != ErrCheckpointFileNotFound {
+			return FileSetFile{}, false, err
+		}
+
+		if isLegacy {
+			pattern = filesetFileForTime(blockStart, anyLowerCaseCharsPattern)
+		} else {
+			pattern = filesetFileForTimeAndVolumeIndex(blockStart, volume, anyLowerCaseCharsPattern)
```

Comment:
I think pattern will be unset if `volume != 0`; this should probably check after the conditionals and set it to `filesetFileForTimeAndVolumeIndex(blockStart, volume, anyLowerCaseCharsPattern)` if it's still empty